outward_assembly/pipeline_steps.py

```diff
@@ -455,6 +458,8 @@ def _subset_split_files_batch(
 
     # Run Nextflow with single dynamic config (which includes static configs)
     nextflow_cmd = ["nextflow", "run", str(nextflow_main), "-c", str(dynamic_config)]
+    if profile := os.environ.get(NF_PROFILE_ENV_VAR):
+        nextflow_cmd.extend(["-profile", profile])
```

Comment:
Rather than checking an env variable, it's possible to use a keyword argument here, then at test time use monkeypatch to replace `_subset_split_files_batch` with a wrapped version that sets the keyword. That makes the code here ever so slightly simpler, but I wonder whether having the env variable could ever be useful.

(The obvious alternative is to just percolate a `batch_profile` keyword arg all the way up to the top level user-facing `outward_assembly` function, which I don't love because that's a lot of percolation for a keyword arg the user should never need to think about or change.)